app/gui/src/project-view/providers/widgetRegistry/configuration.ts

```diff
@@ -217,22 +229,32 @@ export function functionCallConfiguration(
   }
 }
 
+/** Flatten possibly nested choice. */
+export function flattenChoice(choice: Choice): FlattenedChoice[] {
+  if (typeof choice.value === 'string') {
+    return [choice as FlattenedChoice]
+  }
+  return choice.value.flatMap(flattenChoice)
+}
+
 /** A configuration for the inner widget of a single-choice selection widget. */
 export function singleChoiceConfiguration(config: SingleChoice): OneOfFunctionCalls {
+  const possibleChoices = config.values.flatMap(flattenChoice)
   return {
     kind: 'OneOfFunctionCalls',
     possibleFunctions: new Map(
-      config.values.map((value) => [value.value, functionCallConfiguration(value.parameters)]),
+      possibleChoices.map((choice) => [choice.value, functionCallConfiguration(choice.parameters)]),
     ),
   }
 }
 
 /** A configuration for the inner widget of a multiple-choice selection widget. */
 export function multipleChoiceConfiguration(config: MultipleChoice): SomeOfFunctionCalls {
+  const possibleChoices = config.values.flatMap(flattenChoice)
   return {
     kind: 'SomeOfFunctionCalls',
     possibleFunctions: new Map(
-      config.values.map((value) => [value.value, functionCallConfiguration(value.parameters)]),
+      possibleChoices.map((choice) => [choice.value, functionCallConfiguration(choice.parameters)]),
     ),
```

Comment:
It would be supernice to have these method tested.

If they are (I couldn't find), then extend them with case with nested choices.

Reply:
Added some tests, although the implementation is very straightforward and I doubt anything will break here, but it can break in the usage place instead, and it is far more complicated to test.